src/LoanAgent.sol

```diff
@@ -124,14 +206,61 @@ contract LoanAgent {
     // The minerShare won't be withdrawn and the amount will be added in totalMinerCredit which the miner can withdraw anytime
     // update availableBalanceHistory, withdrawHistory and lastWithdrawlCheckpoint
     function withdrawEarnings(
-        uint256 checkpoint,
-        uint256 minerShare,
-        uint256 usersShare,
-        uint256 oracleShare
-    ) external onlyOracle {}
+        uint checkpoint,
+        uint minerShare,
+        uint usersShare,
+        uint oracleShare
+    ) external onlyOracle {
+        CommonTypes.BigInt memory availableBalance = MinerAPI
+            .getAvailableBalance(addressInfo.filActorId);
+        require(
+            !availableBalance.neg,
+            "availableBalance must be positive to withdraw"
+        );
+        MinerAPI.withdrawBalance(addressInfo.filActorId, availableBalance);
+        // TODO: implementation
+    }
 
     // The amount that realOwner can withdraw anytime.
     // The amount must be less than totalOwnerCredit and available balance.
     // The withdrawal will update totalOwnerCredit, availableBalanceHistory
-    function withdrawMinerEarning(uint256 amount) external onlyRealOwner {}
+    function withdrawMinerEarning(uint amount) external onlyRealOwner {
+        accumulateDebt();
+        require(
+            amount <= totalOwnerCredit,
+            "cannot withdraw more than totalOwnerCredit"
+        );
+        (uint availableBalance, bool _neg) = BigInts.toUint256(
```

Comment:
They have these functions? I dreaded my entire day trying to figure this out. BigInt to Uint256 :)